src/components/miscellaneous/TopbarNotifications.tsx

```diff
@@ -0,0 +1,190 @@
+import Linkify from 'components/miscellaneous/Linkify';
+import Pagination from 'components/layout/Pagination';
+import Dialog from 'components/layout/Dialog';
+import { useGoogleAnalytics } from 'hooks/Utils';
+import Paper from 'components/layout/Paper';
+
+type NotificationItemProps = {
+  notification: Notification;
+};
+
+export type NotificationsTopbarProps = {
+  color: React.CSSProperties['backgroundColor'];
+};
+
+const NotificationItem = ({ notification }: NotificationItemProps) => {
+  const [showDescription, setShowDescription] = useState(false);
+  const { event } = useGoogleAnalytics();
+  const updateNotification = useUpdateNotification(notification.id);
+
+  useEffect(() => {
+    return () => {
+      if (!notification.read) {
+        updateNotification.mutate(true);
+      }
+    };
+  }, [notification]);
+
+  const Icon = notification.read ? NotificationReadIcon : NotificationUnreadIcon;
+
+  return (
+    <Paper noOverflow noPadding sx={{ mb: 1, ...(!notification.read && { backgroundColor: (theme) => theme.palette.colors.tihlde + '25' }) }}>
+      <ListItem
+        secondaryAction={
+          <>
+            {notification.link && (
+              <IconButton
+                {...(isExternalURL(notification.link) ? { component: 'a', href: notification.link } : { component: Link, to: notification.link })}
+                aria-label='Åpne link'
+                edge='start'
+                onClick={() => event('open-notification-link', 'notifications', `Opened notification link: ${notification.link}`)}>
+                <LinkIcon />
+              </IconButton>
+            )}
+            {notification.description !== '' && (
+              <IconButton aria-label='Åpne beskrivelse' edge='end' onClick={() => setShowDescription((prev) => !prev)}>
+                {showDescription ? <ExpandLessIcon /> : <ExpandMoreIcon />}
+              </IconButton>
+            )}
+          </>
+        }>
+        <ListItemIcon>
+          <Icon />
+        </ListItemIcon>
+        <ListItemText primary={<Linkify>{notification.title}</Linkify>} secondary={getTimeSince(parseISO(notification.created_at))} />
+      </ListItem>
+      <Collapse in={showDescription}>
+        <Divider />
+        <Typography sx={{ whiteSpace: 'break-spaces', overflowWrap: 'break-word', py: 1, px: 2 }} variant='body2'>
+          <Linkify>{notification.description}</Linkify>
+        </Typography>
+      </Collapse>
+    </Paper>
+  );
+};
+
+const NotificationItemLoading = () => (
+  <Paper noPadding sx={{ mb: 1 }}>
+    <ListItem>
+      <ListItemIcon>
+        <NotificationReadIcon />
+      </ListItemIcon>
+      <ListItemText primary={<Skeleton height={24} width='40%' />} secondary={<Skeleton height={12} width='15%' />} />
+    </ListItem>
+  </Paper>
+);
+
+const NotificationsTopbar = ({ color }: NotificationsTopbarProps) => {
+  const [showNotificationItem, setShowNotificationItem] = useState(false);
```

Comment:
Mer presis navngivning:

```suggestion
  const [showNotifications, setShowNotifications] = useState(false);
```